src/serialization.cpp

```diff
@@ -69,22 +69,22 @@ Status serializeBlobToTensorProto(
     }
     }
     responseOutput.mutable_tensor_shape()->Clear();
-    auto& effectiveNetworkOutputShape = networkOutput->getEffectiveShape();
+    auto& effectiveNetworkOutputShape = networkOutput->getShape_3();
     auto& actualBlobShape = getEffectiveBlobShape(blob);
     if (effectiveNetworkOutputShape.size() != actualBlobShape.size()) {
         SPDLOG_ERROR("Failed to serialize blob: {}. There is difference in number of dimensions expected:{} vs actual:{}",
             networkOutput->getName(), effectiveNetworkOutputShape.size(), actualBlobShape.size());
         return StatusCode::INTERNAL_ERROR;
     }
     for (size_t i = 0; i < effectiveNetworkOutputShape.size(); ++i) {
-        size_t dim;
-        if (effectiveNetworkOutputShape[i] != 0) {
-            if (effectiveNetworkOutputShape[i] != actualBlobShape[i]) {
+        dimension_value_t dim;
+        if (effectiveNetworkOutputShape[i] != DYNAMIC_DIMENSION) {
```

Comment:
Use `effectiveNetworkOutputShape[i].isAny()`
Hide `DYNAMIC_DIMENSION` as much as we can

Reply:
Check if match if yes then use proto

Reply:
Potentially add Shape.match(TensorProto)

Reply:
Shape.match(ov::Tensor)

Reply:
This is already implemented in _2. This code in theory is dead now & will be removed in 2.0 cleanup